_targets/build.project.armv7m7-imxrt106x

```diff
@@ -95,34 +94,36 @@ b_build_target() {
 
 	b_log "Relink plo to include the new script"
 	(cd plo && make $MAKEFLAGS all)
+
+	cp "${PREFIX_PROG_STRIPPED}plo-ram-${TARGET}.img" _boot/
 }
 
 
 b_add2img() {
-	printf "Copying %s (offs=%dB)\n" $1 $OFFSET
-	dd if=$1 of=$2 bs=1 seek=$OFFSET conv=notrunc 2>/dev/null
-	SZ=$((($(stat -c "%s" $1) + $SIZE_PAGE - 1)&$PAGE_MASK))
-	OFFSET=$(($OFFSET+$SZ))
+	printf "Copying %s (offs=%dB)\n" "$1" "$OFFSET"
+	dd if="$1" of="$2" bs=1 seek="$OFFSET" conv=notrunc 2>/dev/null
+	SZ=$((($(stat -c "%s" "$1") + SIZE_PAGE - 1) & PAGE_MASK))
+	OFFSET=$((OFFSET + SZ))
 }
 
 
 b_image_target() {
 	b_log "Creating image"
 
 	IMG=_boot/phoenix-${TARGET}.disk
-	rm -f $IMG
+	rm -f "$IMG"
 
 	# Loader
 	OFFSET=0
-	b_add2img ${PREFIX_PROG_STRIPPED}plo-${TARGET}.img $IMG
+	b_add2img "${PREFIX_PROG_STRIPPED}plo-${TARGET}.img" "$IMG"
 
 	# Kernel
-	OFFSET=$KERNEL_OFFS
-	b_add2img ${PREFIX_PROG_STRIPPED}phoenix-${TARGET}.elf $IMG
+	OFFSET="$KERNEL_OFFS"
+	b_add2img "${PREFIX_PROG_STRIPPED}phoenix-${TARGET}.elf" "$IMG"
 
 	#Programs
 	for prog in "${PROGS[@]}"; do
-		b_add2img ${PREFIX_PROG_STRIPPED}$prog $IMG
+		b_add2img "${PREFIX_PROG_STRIPPED}$prog" "$IMG"
```

Comment:
it seems prone to errors that You compute where the programs will land in flash two times (first for PLO, then for actual placement). It would be better to put images into the image and write the PLO script at the same time.